rabbitmq/resource_queue.go

```diff
@@ -75,6 +84,19 @@ func CreateQueue(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Unable to parse settings")
 	}
 
+	// If json_arguments is used, unmarshal it into a generic interface
+	// and use it as the "arguments" key for the queue.
+	if v, ok := settingsMap["json_arguments"].(string); ok && v != "" {
```

Comment:
as mentioned on Slack, would this make more sense as an interpolation function passed into `arguments`? i.e.

```
arguments = "${json_to_map("{ "hello": "world" })"
```